xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.h

```diff
@@ -0,0 +1,29 @@
+/*
+ *  Copyright (C) 2010-2019 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#import <Foundation/NSDictionary.h>
+
+typedef NS_ENUM(unsigned int, DarwinEmbedPlaybackState) {
+  DARWINEMBED_PLAYBACK_STOPPED,
+  DARWINEMBED_PLAYBACK_PAUSED,
+  DARWINEMBED_PLAYBACK_PLAYING
+};
+
+@interface DarwinEmbedNowPlayingInfoManager : NSObject
+
+@property(readwrite) NSDictionary* nowPlayingInfo;
+@property(readwrite) DarwinEmbedPlaybackState playbackState;
+
+- (void)setDarwinEmbedNowPlayingInfo:(NSDictionary*)info;
```

Comment:
if this is the intended way to set `nowPlayingInfo` property, there're 2 ways to achieve the same behavior without additional method:

1. override property setter and perform custom logic there
2. use KVO on the property and perform custom logic
3. (without removing this method, not recommended) declare property in the header as `readonly` and re-declare it as writeable in the .mm

Reply:
Went the override route.